extensions/notebook/src/book/bookTocManager.ts

```diff
@@ -470,8 +469,10 @@ export class BookTocManager implements IBookTocManager {
 		};
 
 		if (isSection) {
+			await fs.mkdir(path.dirname(pathDetails.filePath));
 			fileEntryInToc.sections = [];
 		}
+		await fs.writeFile(pathDetails.filePath, '');
```

Comment:
What is the file that's being written when we're adding a section? Wouldn't a section just be a folder?

Reply:
Currently, when we load a book we need a file or url on every toc entry and if we want to make it a section we add the `sections` keyword in the toc.

Reply:
So if I have a section `/path/to/sectionfoo`

Then we'll end up with both a folder named `sectionfoo` and inside that a file also named `sectionfoo`?

Reply:
We let the user enter the file name in the dialog, so it should be `/path/to/sectionfoo/file_name`